Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -244,9 +244,11 @@ def audit_style
     def audit_file
       # Under normal circumstances (umask 0022), we expect a file mode of 644. If
       # the user's umask is more restrictive, respect that by masking out the
-      # corresponding bits. (The also included 0100000 flag means regular file.)
-      wanted_mode = 0100644 & ~File.umask
+      # corresponding bits. (The also-included 0100000 flag means regular file.)
+      # We do not care about the group write bit here, since some brew setups
+      # (e.g., brewdo) make it desirable for it to be set.
       actual_mode = formula.path.stat.mode
+      wanted_mode = (0100644 | (actual_mode & 0020)) & ~File.umask
```

Comment:
Can this be simplified a bit? It's a bit hairy as-is.

Reply:
Yeah, it is... Sadly, I don't think so. The obvious way to understand it is that `actual_mode & 0020` select the group write bit from the actual file mode (i.e., gives either `0020` or `0000`) and then sets the corresponding bit on `0100644` if so. i.e., it takes `0100644` and adds the group write bit *if and only if* the group write bit is set on the actual fine -- which makes it `0100664` instead. I think it's right that `& ~File.umask` should remain as the outermost bit operation, right? Anyway, if you want me to put another comment in for this, or rework it into a form you have in mind, let me know. :-)

Reply:
What about just checking the individual bits?

Reply:
@MikeMcQuaid Yep, that occurred to me too, but `wanted_mode` is used for the diagnostic message just below, which makes this calculation quite convenient I think.

Reply:
I think the message should change too. It's no longer particularly accurate if we're going to ignore the group bit.

Reply:
Fair enough. At the moment it obviously it "conservatively" suggests new permissions by reflecting whatever the group write bit already is. What sort of message would you like to see?

Reply:
Probably just two separate messages for world read and only user write?

Reply:
Okay, sounds fair. I'll see what I can do now.

Reply:
Wait, so to confirm... 
* We *want* user-write (i.e., error if not set).
* We don't care whether or not group-write is set (i.e., never error on this).
* We *do not want* other-write (i.e., error if set).

Reply:
Yep, sounds right to me!

Reply:
Great. This is what I implemented. I'll just adjust the tests now and push.

Reply:
Ah, the problem is now: what if the above conditions are all satisfied, but the umask is more restrictive? Should we honour that or not? If so, it means a 4th problem message potentially, which awkward, since it would effectively make the previous ones redundant.

Reply:
(Per the above, if we want to respect the umask, I think my previous approach works best. Otherwise, we can go with yours.)

Reply:
I'm not sure I fully understand how we currently honour or do not the umask. Can you explain a case in which it would affect an audit failure and what your opinion is on what we should do? Thanks ❤️

Reply:
As I understand, the current behaviour is basically that the umask is used to "mask out" (clear) certain bits when incorrect permissions are found and a chmod is suggested (with a numeric rather than a symbolic argument). The Wikipedia page on umask has an interesting sentence:

> Generally, the mask only affects file permissions during the creation of new files and has no effect when file permissions are changed in existing files, however, the chmod command will check the mask when the chmod options are specified using symbolic mode and a reference to a class of users is not specified.

Obviously, we are not dealing with the creation of files here. The question is, is our situation analagous to a symbolic change without a reference to a class of users (i.e., something like `+r` as opposed to `g-w`, the latter making a reference to the class of users `g`). I would say no, so that we can probably just not worry about the umask. What do you think?

Reply:
> I would say no, so that we can probably just not worry about the umask. What do you think?

Agreed!